src/Events/Controllers/CollectController.cs

```diff
@@ -77,7 +77,15 @@ public async Task<IActionResult> Post([FromBody] IEnumerable<EventModel> model)
                     }
                     if (cipher == null)
                     {
-                        continue;
+                        // When the user cannot access the cipher directly, check if the organization allows for
+                        // admin/owners access to all collections and the user can access the cipher from that perspective.
+                        var org = await _organizationRepository.GetByIdAsync(eventModel.OrganizationId.Value);
+                        cipher = await _cipherRepository.GetByIdAsync(eventModel.CipherId.Value);
+
+                        if (!org.AllowAdminAccessToAllCollectionItems || cipher == null)
```

Comment:
⛏️ We should also have a "sanity" check to be sure the returned cipher belongs to the organization we checked. If they don't match something is wrong and we should skip.

Reply:
👍 [7fbd875](https://github.com/bitwarden/server/pull/4705/commits/7fbd875a6e01ba1d150df36275412174fc0d902e)